worlds/wargroove2/__init__.py

```diff
@@ -0,0 +1,191 @@
+from .Rules import set_rules
+from worlds.AutoWorld import World, WebWorld
+from .Options import wargroove2_options
+from worlds.LauncherComponents import Component, components, Type, launch_subprocess
+
+
+def launch_client():
+    from .client import launch
+    launch_subprocess(launch, name="Wargroove2Client")
+
+
+components.append(Component("Wargroove 2 Client", "Wargroove2Client", func=launch_client, component_type=Type.CLIENT))
+
+
+class Wargroove2Settings(settings.Group):
+    class RootDirectory(settings.UserFolderPath):
+        """
+        Locate the Wargroove 2 root directory on your system.
+        This is used by the Wargroove 2 client, so it knows where to send communication files to
+        """
+        description = "Wargroove 2 root directory"
+
+    root_directory: RootDirectory = RootDirectory("C:/Program Files (x86)/Steam/steamapps/common/Wargroove 2")
+
+
+class Wargroove2Web(WebWorld):
+    tutorials = [Tutorial(
+        "Multiworld Setup Guide",
+        "A guide to setting up Wargroove 2 for Archipelago.",
+        "English",
+        "wargroove2_en.md",
+        "wargroove2/en",
+        ["Fly Sniper"]
+    )]
+
+    options_presets = wargroove2_option_presets
+
+
+class Wargroove2World(World):
+    """
+    Command an army, in the sequel to the hit turn based strategy game Wargroove!
+    """
+
+    option_definitions = wargroove2_options
+    settings: typing.ClassVar[Wargroove2Settings]
+    game = "Wargroove 2"
+    topology_present = True
+    data_version = 1
+    web = Wargroove2Web()
+    level_list: [Wargroove2Level] = None
+    first_level: Wargroove2Level = None
+    final_levels: [Wargroove2Level] = None
+
+    item_name_to_id = {name: data.code for name, data in item_table.items()}
+    location_name_to_id = location_table
+
+    def _get_slot_data(self):
+        return {
+            'seed': "".join(
+                self.random.choice(string.ascii_letters) for i in range(16)),
+            'income_boost': self.options.income_boost.value,
+            'commander_defense_boost': self.options.commander_defense_boost.value,
+            'starting_groove_multiplier': self.options.groove_boost.value,
+            'level_shuffle_seed': self.options.level_shuffle_seed.value,
+            'can_choose_commander': self.options.commander_choice.value != 0,
+            'final_levels': self.options.final_levels.value,
+            'death_link': self.options.death_link.value == 1
+        }
+
+    def generate_early(self):
+        # First level
+        self.first_level = get_first_level(self.player)
+
+        # Standard levels
+        self.level_list = get_level_table(self.player)
+        low_victory_checks_levels = list(level for level in self.level_list if level.low_victory_checks)
+        high_victory_checks_levels = list(level for level in self.level_list if not level.low_victory_checks)
+        if self.multiworld.level_shuffle_seed[self.player] == 0:
+            random = self.multiworld.random
```

Comment:
```suggestion
            random = self.random
```
You should use your world's random object rather than the multiworlds -- it makes it more deterministic.